### PR TITLE
migrate createLock to ethers

### PR DESCRIPTION
**Pull request by cellog**
# Description

this is our first contract method migrated to ethers

# Issues

Refs #2782 

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [X] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [X] My code follows the style guidelines of this project, including naming conventions
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [X] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-js/src/__tests__/helpers/walletServiceHelper.ethers.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import * as UnlockV01 from 'unlock-abi-0-1'
 import * as UnlockV02 from 'unlock-abi-0-2'
 import { ethers } from 'ethers'
 
-import * as utils from '../../utils'
+import * as utils from '../../utils.ethers'
```
Comment (cellog) [inline unlock-js/src/__tests__/helpers/walletServiceHelper.ethers.js]: use the utils from this PR

File: unlock-js/src/__tests__/helpers/walletServiceHelper.ethers.js [hunk 2]
```diff
@@ -32,10 +32,10 @@ export const prepWalletService = async (
   switch (contract) {
     case UnlockV01.Unlock:
     case UnlockV01.PublicLock:
-    case UnlockV02.PublicLock: // 1 is 2
       unlockVersion =
         '0x0000000000000000000000000000000000000000000000000000000000000001'
       break
+    case UnlockV02.PublicLock:
```
Comment (cellog) [inline unlock-js/src/__tests__/helpers/walletServiceHelper.ethers.js]: the last PR fixed this issue!

File: unlock-js/src/walletService.js [hunk 3]
```diff
@@ -100,6 +100,31 @@ export default class WalletService extends UnlockService {
     return Promise.resolve(address)
   }
 
+  /**
```
Comment (cellog) [inline unlock-js/src/walletService.js]: this is the ethers equivalent of `_sendTransaction`

**Comment by cellog** [comment]
I'm going to do a separate PR with the walletService addition including tests, and rebase this on it

File: unlock-js/src/utils.ethers.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { utils, constants } from 'ethers'
+
+// this allows us to flexibly upgrade web3 and fix bugs as they surface
+// or to migrate to a totally different library and have a single point of modification
+module.exports = {
+  toWei: (value, units) => utils.parseUnits(value, units),
+  hexlify: utils.hexlify,
+  hexStripZeros: utils.hexStripZeros,
+  bigNumberify: utils.bigNumberify,
+  ZERO: constants.AddressZero,
```
Comment (julien51) [inline unlock-js/src/utils.ethers.js]: Nit: I'd move this to `constants.js`?

Comment (cellog) [reply unlock-js/src/utils.ethers.js]: ok

**Pull request merged**